x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -165,9 +165,13 @@ export const createInventoryMetricThresholdExecutor =
     }
     const source = await libs.sources.getSourceConfiguration(savedObjectsClient, sourceId);
 
-    const [, { logsShared }] = await libs.getStartServices();
+    const [, { logsShared, logsDataAccess }] = await libs.getStartServices();
+
+    const logSourcesService =
+      logsDataAccess.services.logSourcesServiceFactory.getLogSourcesService(savedObjectsClient);
+
     const logQueryFields: LogQueryFields | undefined = await logsShared.logViews
-      .getClient(savedObjectsClient, esClient)
+      .getClient(savedObjectsClient, esClient, logSourcesService)
```

Comment:
I understand why we changed the log threshold executor, but can you please help me understand why we also need to change the metric threshold executor? Also, does the same apply to the inventory rule?

Reply:
Sure - the line you've highlighted creates a log views client, this is needed to gather the indices to query for logs (the `source` provides the metric ones). This is due to the inventory metric threshold rule allowing the selection of a `logRate` metric, [when using `logRate` the logs indices are used](https://github.com/elastic/kibana/blob/0eea85e86937edd914be43ed8f6efeb038966f23/x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/lib/get_data.ts#L141) to evaluate the data in the executor. 

I only found a reference in the inventory metric threshold rule, not the metric threshold one.

Reply:
Ah, this is inventory rule, not metric threshold (The naming always confuses me!)

So, if we select log rate in the inventory rule, we will use the insides from the log app instead of infra, right? I would have never guessed that 😅 

![image](https://github.com/user-attachments/assets/f8273f8f-39d8-4370-931b-326fc3c5893e)

Reply:
Yep, selecting that metric will query logs indices.